[FE] 프로필 페이지 - 멘토

ChanYangYu opened the pull request:
## 관련 이슈

- closes #69

## 셀프 체크리스트

> 최소 1명 이상의 assign을 받아야만 Merge 가능합니다.

- [X] Warning Message 발생 유/무
- [X] Coding Convention 준수 유/무

## 작업 요약
> 작업 내역 요약
- 멘토링 스택, 멘토링 신청, 멘토링 요청 리스트 모달 등 남은 기능 모두 구현
- 다른 유저 프로필 보기 기능 구현완료
## Preview

> 선택 사항


## PR 특이 사항

https://user-images.githubusercontent.com/55623688/142028982-937ef350-50b4-4185-86c4-02b7b9207e1c.mp4


> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점
> 백로그에 없지만 필요하여 구현한 기능 

- 프로필페이지가 별거없어보였는데... 하면할수록 신경써야할게 산더미라서 오늘 겨우 마무리할 수 있었습니다..
- 프론트엔드는 건드린게 많아서 영상을 첨부했습니다! 백엔드가 된다고 가정하고 작성하였기 때문에 #120 번 PR을 먼저 머지후에 머지해야 정상동작합니다!

ChanYangYu committed:
[FE] refactor: MentorStackBox 리팩토링


- Props로 두개의 모달을 제어할 수 있는 함수를 받도록 수정
- profileDataSlice에서 관리하는 isMentor 변수를 통해 멘토신청 박스를 보여줄지 멘토링 스택을 보여줄지 판단하도록 로직 수정
- issue (#69)

ChanYangYu committed:
[FE] style: Global Style에 색상 추가 및 적용  


- Error : 붉은 색
- PrimaryHover: Primary버튼 Hovering 색상

ChanYangYu committed:
[FE] refactor: formatDateToString 수정 및 코드 적용


- formatDateToString에 string 타입을 추가하고 ProfileActivityListBox에서 사용하도록 코드 로직 변경

ChanYangYu committed:
[FE] feat: RequestMentoring 모달 구현


- RequestMentoring 모달 구현
- 임시적으로 CreateMentoringStack을 통해 isMentor를 수정하면 멘토링 스택페이지가 나오도록 변경
- issue (#69)

ChanYangYu committed:
[FE] feat: mentor여부 체크 서버 확인 API 연결


- mentorId 정보 API 연결
- MentorInfoResponse 타입 생성
- updateTechStackRequest -> UpdateTechStackRequest 
- issue (#69)

ChanYangYu committed:
[FE] feat: 멘토신청 POST API 연결


- RegisterMentorRequest 서버 전송 데이터 타입 생성
- 멘토 신청 서버 요청 로직 구현
- issue (#69)

ChanYangYu committed:
[FE] feat: 멘토링 스택 서버 데이터 연결


- mentoring Stack fetch 로직 구현
- issue (#69)

ChanYangYu committed:
[FE] feat: fetch 부분 코드 리팩토링


- 모든 fetch 과정 ProfilePage index에서 관리하도록 변경하여 fetch가 될때까지 loader를 뛰우도록 로직 수정
- 문제 !  5개의 fetch요청이 await으로 연결되어 있어 거의 동기적으로 요청하기 때문에 요청시간이 꽤 긴 문제가 있다. (약, 0.5초)
- issue (#69)

ChanYangYu committed:
[FE] style: 로더 디자인 변경


- 로더 디자인 변경

ChanYangYu committed:
[FE] rename:  ServerData 타입 이름 변경


- 타입 뒤에 Request붙는 타입 이름 변경 (MentoringRequest 이름이랑 헷갈리는 경우 발생)

ChanYangYu committed:
[FE] feat: 멘토링 요청 리스트 서버 데이터 연결


- 멘토링 요청 리스트 API 데이터 연동
- issue (#69)

ChanYangYu committed:
[FE] feat: 멘토링 요청 수락 거절 구현


- 멘토링 요청 수락/거절 API 연결

ChanYangYu committed:
[FE] design: ActivityListItem 디자인 변경


- ActivityListItem 마진 및 출력 포맷 수정

who-is-hu reviewed (changes requested):
본인이 아닌 다른 사용자의 프로필 페이지 들어갈때 상황도 고려해서
기능을 추가해야할것 같습니다.

ChanYangYu committed:
[FE] refactor: 프로필페이지 데이터 fetch 리팩토링


- fetch 함수 모두 FetchProfileData.ts로 분리
- 유저기본 정보(깃허브 프로필정보, 유저id, 열정,나눔스택) 로그인정보를 사용하지 않고 URL 깃허브 아이디를 사용하여 API 요청하여 데이터 출력하도록 변경
- issue (#69)

yuchanleeme commented on client/src/pages/ProfilePage/index.tsx:
```diff
@@ -4,31 +4,101 @@ import ProfileSideContents from './ProfileSideContents';
+      const fetchedGroupActivity = await groupHttpClient.getGroupActivity(userId);
+      const groupActivitys = fetchedGroupActivity.map(({name, startAt, endAt}) =>{
+        return {
+          name: name,
```
여기도 `name:name,` -> `name,` 으로 해도 좋을것 같습니다!

yuchanleeme commented on client/src/pages/ProfilePage/index.tsx:
```diff
@@ -4,31 +4,101 @@ import ProfileSideContents from './ProfileSideContents';
+        await fetchProfileIntro(userId);
+        await fetchProfileTechStack(userId);
+        await fetchGroupActivityTechStack(userId);
+
```
다 함수로 분리해서 코드가 이뻐진것 같아 너무 좋네요!! 에러 잡는 코드 적는것도 잘하신것 같습니다!! 👍

ChanYangYu replied:
오...  try...catch를 리팩토링하면서 다시 빼먹었는데 덕분에 다시 넣었습니다 ㅎㅎ

yuchanleeme reviewed (approved):
전체 로직 확인했습니다!! 수고하셨어요~~!!

ChanYangYu committed:
[FE] feat: 다른 유저 프로필 정보 출력


- 다른 유저 프로필의 경우 편집기능이 안되도록 수정
- 로그인 정보 기반의 데이터에서 전역 ProfileData로 화면 데이터 출력
- issue (#69)

ChanYangYu committed:
[FE] fix: URL 이동 문제 해결


- 다른 사람 프로필페이지에서 헤더의 버블 모달로 내 프로필 이동시 url만 바뀌고 정상적으로 화면 리랜더링이 되지 않는 문제 해결
  -> router-dom의 param을 설정하여 리랜더링하도록 변경
- 없는 아이디로 URL을 통해 접근시 메인화면으로 리다이렉트

Donggggg commented:
> 본인이 아닌 다른 사용자의 프로필 페이지 들어갈때 상황도 고려해서 기능을 추가해야할것 같습니다.

해당 케이스만 처리하면 좋을 것 같아요!

ChanYangYu committed:
[FE] style: 피드백 적용 


- { name:name, ... }, -> { name, ... }
- fetch 코드 : try... catch 적용

ChanYangYu committed:
[FE] fix: loader toggle state 에러 해결


- toggle state에서 비동기 처리가 되면서 3번 실행될 경우 로더가 사라지지 않는 문제 발생
  -> utilSlice에서 직접 로더의 상태를 설정할 수 있는 reducer 생성
